src/main/kotlin/org/rust/lang/core/resolve/ImplLookup.kt

```diff
@@ -472,6 +475,12 @@ class ImplLookup(
             return TyFunction(argumentTypes, outputType)
         }
 
+    fun isTraitVisibleFrom(trait: RsTraitItem, scope: RsElement): Boolean {
+        val name = trait.name ?: return true
+        val path = RsPsiFactory(project).tryCreatePath(name)?.apply { setContext(scope) } ?: return true
+        return resolvePath(path, this).any { it.element == trait }
```

Comment:
I wonder if these uncaheble resolves have any impact on performance.

Reply:
This code anyway executes very rarely - only if we run into method name ambiguity (resolved 2 methods with the same name). I've checked it with our `the best` :) performance tests and did not noticed any regression

Reply:
Yeah, a fair point about this being rare!

I foresee though that one day we'll start to use this for "import trait" quick fix, and it'll become a bottleneck :) But we definitely should not try to optimize it right now!